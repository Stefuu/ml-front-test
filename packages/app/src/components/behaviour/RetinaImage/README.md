# RetinaImage

## Shows regular or retina image version, depending on user screen support
### The difference can be observed using chrome emulation tool to toggle the pixel ratio
