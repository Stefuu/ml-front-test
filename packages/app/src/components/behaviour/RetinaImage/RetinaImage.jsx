import styled from 'styled-components'
import { height, space, width } from 'styled-system'

const RetinaImage = styled.div`
    background-image: url(${props => props.regular});
    background-size: contain;
    background-repeat: no-repeat;
    @media 
    (-webkit-min-device-pixel-ratio: 2), 
    (min-resolution: 192dpi) { 
        background-image: url(${props => props.retina});
    }
    ${height}
    ${space}
    ${width}
`

export default RetinaImage
