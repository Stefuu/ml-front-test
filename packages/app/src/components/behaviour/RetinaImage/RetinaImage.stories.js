import React from 'react'
import { storiesOf } from '@storybook/react'
import { doc } from 'storybook-readme'
import readmeFile from './README.md'
import centered from '@storybook/addon-centered'
import { withKnobs } from '@storybook/addon-knobs'
import theme from '../../themes/default'
import { ThemeProvider } from 'styled-components'
import RetinaImage from './RetinaImage'
import LogoImageRegular from '../../../assets/Logo_ML.png'
import LogoImageRetina from '../../../assets/Logo_ML@2x.png'

storiesOf('app/components/behaviours/RetinaImage', module)
  .addParameters({ jest: ['RetinaImage.test.js'] })
  .add('README', doc(readmeFile))
  .addDecorator(centered)
  .addDecorator(withKnobs)
  .add('simple', () => (
    <ThemeProvider theme={theme}>
      <RetinaImage
        width={53}
        height={36}
        regular={LogoImageRegular}
        retina={LogoImageRetina}
        alt='Logo' />
    </ThemeProvider>
  ))
