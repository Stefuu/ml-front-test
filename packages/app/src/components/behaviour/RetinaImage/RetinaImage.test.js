import React from 'react'
import { render } from 'react-testing-library'
import RetinaImage from './RetinaImage'
import LogoImageRegular from '../../../assets/Logo_ML.png'
import LogoImageRetina from '../../../assets/Logo_ML@2x.png'

describe('RetinaImage', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(<RetinaImage
      width={53}
      height={36}
      regular={LogoImageRegular}
      retina={LogoImageRetina}
      alt='Logo' />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
