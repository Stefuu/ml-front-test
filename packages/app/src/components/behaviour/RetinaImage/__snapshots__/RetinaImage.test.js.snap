// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetinaImage Should render [snapshot] 1`] = `
<div
  alt="Logo"
  class="sc-bdVaJa kXMFPb"
  height="36"
  width="53"
/>
`;
